pkg/op/server_legacy.go

```diff
@@ -205,10 +205,15 @@ func (s *LegacyServer) CodeExchange(ctx context.Context, r *ClientRequest[oidc.A
 	if err != nil {
 		return nil, err
 	}
+
 	if r.Client.AuthMethod() == oidc.AuthMethodNone {
```

Comment:
Maybe there is no more need for this if checking if CodeVerifier exists instead.

Reply:
I would keep both checks, as with `oidc.AuthMethodNone` the check must be enforced. However, we can use a OR here for both conditions.

```go
if r.Client.AuthMethod() == oidc.AuthMethodNone || r.Data.CodeVerifier != "" {
```